x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -281,6 +281,7 @@ func (k Keeper) GetLightClientTimeSafely(ctx sdk.Context, connectionID string) (
 	conn, found := k.IBCKeeper.ConnectionKeeper.GetConnection(ctx, connectionID)
 	if !found {
 		k.Logger(ctx).Error(fmt.Sprintf("invalid connection id, \"%s\" not found", connectionID))
+		return 0, false
```

Comment:
Glad we patched this one, the LC was updating on line 287 even when connection id was invalid

Reply:
😬

Reply:
Unfort I don't think it was the cause of the 50+ LC update txs per block issue, that seems to be more a function of either (a) having too many LCs or (b) updating all LCs on a fixed cadence (e.g. epochly)
I don't have a good answer for it yet [](edited-with-Axolo)

Reply:
Might be a mix of both? Not sure it's an issue [](edited-with-Axolo)